lib/matplotlib/text.py

```diff
@@ -2027,7 +2029,12 @@ def draw(self, renderer):
         # docstring inherited
         if renderer is not None:
             self._renderer = renderer
-        if not self.get_visible() or not self._check_xy(renderer):
+
+        visible = self.get_visible() and self._check_xy(renderer)
+
+        self._check_xytext()
```

Comment:
I think this needs to go after `update_positions` below, since `update_positions` sets the correct transform.